components/client/upload-button.tsx

```diff
@@ -18,23 +21,149 @@ const VisuallyHiddenInput = styled("input")({
+    if (!files) {
+      setErrorMessage("No data selected.");
+      return false;
+    }
+
+    if (emptyLength === files.length) {
+      setErrorMessage(
+        "No files selected. Please choose at least one CSV file.",
+      );
+      return false;
+    }
+
+    const validFiles = Array.from(files).filter(
+      (file: Readonly<File>) =>
+        "text/csv" === file.type || file.name.endsWith(".csv"),
+    );
+
+    if (emptyLength === validFiles.length) {
+      setErrorMessage("Please select at least one valid CSV file.");
+      return false;
+    }
+
+    if (validFiles.length !== files.length) {
+      setErrorMessage("Some files were not valid CSV files.");
+      return false;
+    }
+    return true;
+  };
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>): void => {
+    const { files } = e.target;
+
+    if (isInputValid(files)) {
+      setSelectedFiles(files);
+    } else {
+      setIsUploadFail(true);
+    }
+
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+  };
+
+  useEffect(() => {
+    const uploadFiles = async (): Promise<void> => {
+      if (!selectedFiles) {
+        return;
+      }
+
+      setIsLoading(true);
+
+      try {
+        // Todo: Add a real token in the api call.
+        const token = "";
+        const formData = new FormData();
+        Array.from(selectedFiles).forEach((file: Readonly<File>) => {
+          formData.append("csv-file", file);
+        });
+
+        const response = await fetch(
+          `${process.env.NEXT_PUBLIC_BACKEND_API_HOST}/csv-upload`,
+          {
+            method: "POST",
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+            body: formData,
+          },
+        );
+
+        if (!response.ok) {
+          setErrorMessage(
+            `File upload failed on the server. Status code: ${response.status}.`,
+          );
+          setIsUploadFail(true);
+        } else {
+          setIsUploadSuccess(true);
+        }
+      } catch {
+        setErrorMessage("Failed to connect to the server.");
+        setIsUploadFail(true);
+      } finally {
+        setIsLoading(false);
+        setSelectedFiles(null);
+      }
+    };
+
+    void uploadFiles();
+  }, [selectedFiles]);
+
+  const handleCloseSnackbar = (): void => {
+    setErrorMessage("");
+    setIsUploadSuccess(false);
+    setIsUploadFail(false);
+  };
+
   return (
-    <Button
-      component="label"
-      role={undefined}
-      variant="contained"
-      tabIndex={-1}
-      startIcon={<CloudUploadIcon />}
-    >
-      Upload Orders
-      <VisuallyHiddenInput
-        type="file"
-        onChange={(event) => {
-          console.log(event.target.files);
-        }}
-        multiple
-      />
-    </Button>
+    <>
+      <Button
+        component="label"
+        role={undefined}
+        variant="contained"
+        tabIndex={-1}
+        startIcon={<CloudUploadIcon />}
+        disabled={isLoading}
```

Comment:
@MadsHeilmann Would be better to have some kind of spinner or other loader instead of disabling the button

Reply:
I will open an issue on this as well.